dlib/dnn/loss.h

```diff
@@ -1397,12 +1397,34 @@ namespace dlib
 
             out << "detector_windows:(";
             auto& opts = item.options;
-            for (size_t i = 0; i < opts.detector_windows.size(); ++i)
+
             {
-                out << opts.detector_windows[i].width << "x" << opts.detector_windows[i].height;
-                if (i+1 < opts.detector_windows.size())
-                    out << ",";
+                // write detector windows grouped by label
+                // example output: detector_windows:(aeroplane:74x30,131x30,70x45,54x70,198x30;bicycle:70x57,32x70,70x32,51x70,128x30,30x121;car:70x36,70x60,99x30,52x70,30x83,30x114,30x200)
+                typedef std::deque<const dlib::mmod_options::detector_window_details*> detector_windows;
+                std::map<std::string, detector_windows> detector_windows_by_label;
```

Comment:
This would be simpler if declared like this:
```
std::map<std::string,std::vector<mmod_options::detector_window_details>> detector_windows;
```
No pointers needed, and so no reasoning about lifetimes.

Reply:
Wanted to avoid making unnecessary copies, and also relying on there being a working copy constructor. But agree that simpler is better – changed now.

(At any rate no need to worry about lifetimes I think, because these pointers clearly have a shorter life.)